### PR TITLE
Upgrade React to 16 for dev & tests

### PR DESCRIPTION
**Pull request by wcjordan**
## Description

Upgrade React to React 16 in our dev & test envs.  Also updated Sinon and switched to the default sandbox.

## Motivation and Context

Want to start using React 16 more, so figured a start was to up our dev env.

## How Has This Been Tested?

Tests & site-dev-server

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: test/ReactTouchHandler-test.js [hunk 1]
```diff
@@ -1,46 +1,34 @@
...
 import ReactTouchHandler from '../src/ReactTouchHandler';
 import sinon from 'sinon';
 import { assert } from 'chai';
-import { createRenderer, isElement } from 'react-addons-test-utils';
 
 describe('ReactTouchHandler', function() {
-  var clock, sandbox, requestAnimationFramePolyfillSpy;
-
-  before(function () {
-    clock = sinon.useFakeTimers();
```
Comment (wcjordan) [inline test/ReactTouchHandler-test.js:15]: since all tests share a sandbox I moved this down to the test which actually uses it.

**Comment by ivadenis** [comment]
Is there an update on this?

File: test/ReactTouchHandler-test.js (hunk 1 above)
Comment (pradeepnschrodinger) [reply test/ReactTouchHandler-test.js:15]: nit: clock is defined below in the actual test, so we don't need the declaration here.

File: package.json [hunk 2]
```diff
@@ -46,14 +46,14 @@
     "null-loader": "3.0.0",
     "postcss": "7.0.16",
     "postcss-custom-properties": "6.3.1",
-    "react": "15.6.2",
-    "react-addons-test-utils": "15.6.2",
+    "react": "16.13.1",
```
Comment (pradeepnschrodinger) [inline package.json:49]: So this means we can also remove the `lifecycleCompatibilityPolyfill` which is only used [here](https://github.com/schrodinger/fixed-data-table-2/blob/master/src/FixedDataTableCell.js#L349) to support `getDerivedStateFromProps`?

File: test/ReactTouchHandler-test.js (hunk 1 above)
Comment (wcjordan) [reply test/ReactTouchHandler-test.js:15]: good catch, will do

**Comment by wcjordan** [comment]
@ivadenis deprecated lifecycle methods will be more complicated and may require breaking functionality so it will be handled separately.  It's tracked w/ #512 if you'd like more details.

**Pull request merged**